common/types/chain.go

```diff
@@ -9,6 +9,9 @@ type Sequence interface {
 	Int64() int64 // needed for numeric sequence confirmation - to be removed with confirmation logic generalization: https://smartcontract-it.atlassian.net/browse/BCI-860
```

Comment:
Why can't we just add a new function here as:
Next() Sequence

This will automatically take as input the current Seq, and return the next Seq.

Reply:
That was the original approach but think Jordan brought up some issues with it in his comments above

Reply:
If the generics didn't seem messy then I wouldn't mind going with a method to see how it works out, but e.g. a purely random nonce would seem odd in that model, since it would require you to jump through hoops to wrap up and attach your random generator func as a method on the nonce type.

Pehaps an even more flexible approach would be to accept a pure generator signature of `Next func() NONCE`, which the implementation may pass a pure func, closure, method, helper func using state, etc. - We don' necessarily need to worry about that in this PR though.

Reply:
I've reverted the changes back to the original but updated the name from `Increment` to `Next`. I think in the above scenario, we should be able to create a new sequence for the chain that needs a random nonce and write in the random generator logic directly into the `Next` method. Although we would run into issues if any external parameters are needed for generation.

Reply:
>  I think in the above scenario, we should be able to create a new sequence for the chain that needs a random nonce and write in the random generator logic directly into the Next method. 

This is precisely the scenario I was trying to avoid. We are forcing chains with simple models to build complex types just to fit our type system. There is no need to be this inflexible.

Reply:
This might be me still getting acclimated on the team but I was assuming the expectation was every chain would define its own sequence type like Nonce for EVM. So it would be reasonable having it define how to generate the next sequence.

Reply:
Even if that is the case, we should not be writing code that makes any assumptions about the broader system as a whole, and should choose flexibility by default.

However, the current state was meant to be temporary. We wanted to use pure `fmt.Stringer`, and intend to remove `Int64` with that linked story. If we add more method here, then we are stuck with custom types instead of just plugging things in.

Reply:
I think let us for now, keep it simple, where we get the next nonce from the previous nonce.
For more complex chains like Bitcoin, we can revisit this interface later.

Reply:
This is not about handling complex cases, it is purely about the type system. This method is the more complex choice, and the story we already have to track simplifying this type will not be achievable any longer. I don't see any reason to tie our hands like this - it is just an unnecessary constraint.